docs/resources/azure_virtual_network.md.erb

```diff
@@ -0,0 +1,167 @@
+---
+title: About the azure_virtual_network Resource
+platform: azure
+---
+
+# azure\_virtual_network
+
+Use the `azure_virtual_network` InSpec audit resource to ensure that a Virtual Network (vNet) has been provisioned correctly.
+
+## Syntax
+
+The name of the vNet and the resource group are required as properties to the resource.
+
+    describe azure_virtual_network(group_name: 'MyResourceGroup', name: 'MyNetworkName') do
+      its('property') { should eq 'value' }
+    end
+
+where
+
+* `MyNetworkName` is the name of the virtual network as seen in Azure
+* `MyResourceGroup` is the name of the network's resource group
+* `property` is one of the resource properties
+* `value` is the expected output from the matcher
+
+## Examples
+
+The following examples show to use this InSpec audit resource.
+
+Check that the vNet is of the correct type and that it is in the correct location. 
+Chheck that the vNet has the correct prefix.  Lastly, check that the virtual network has at least 1 subnet attached to it.
+
+    describe azure_virtual_network(group_name: 'InSpec-Azure', name: 'vnet-01') do
+
+         # Check that this named resource is indeed a virtual network
+        its('type') { should cmp 'Microsoft.Network/virtualNetworks' }
+        its('location') { should cmp 'westeurope' }
+
+
+        # Ensure that the address prefix for the VNet is correct
+        # This will return an array so the `include` matcher needs to be used to
+        # see if the specified address prefix is present
+        its('address_prefixes') { should include '10.1.1.0/24' }
+
+        # There should be one subnet
+        its('subnets_count') { should eq 1 }
+
+    end
+
+<br>
+
+## Parameters
+
+* `group_name`, `name`, `apiversion`
+
+## Parameter Examples
+
+The options that can be passed to the resource are as follows.
+
+### group_name (required)
+
+Use this parameter to define the Azure Resource Group to be tested.
+
+    describe azure_virtual_network(group_name: 'InSpec-Azure') do
+        ...
+    end
+
+### name
+
+Use this parameter to define the name of the Azure resource to test.
+
+    describe azure_virtual_network(group_name: 'InSpec-Azure', name: 'vnet-01') do
+        ...
+    end
+
+### apiversion
+
+The API Version to use when querying the resource. Defaults to the latest version for the resource.
+
+    describe azure_virtual_network(group_name: 'InSpec-Azure', name: 'vnet-01', apiversion: '2.0') do
+        ...
+    end
+
+These options can also be set using the environment variables:
+
+* `AZURE_RESOURCE_GROUP_NAME`
+* `AZURE_RESOURCE_NAME`
+* `AZURE_RESOURCE_API_VERSION`
+
+When the options have been set as well as the environment variables, the environment variables take priority.
+
+<br>
+
+## Properties
+
+* `id`,`type`, `location`, `name`, `address_prefixes`, `has_dns_servers`, `dns_servers_count`, `has_subnets`, `subnets_count`, `connected_subnets`
```

Comment:
Super nitpick: id and type don't have a space separator like the rest of the list does.